File: Applied Course/4.Problem Solving/2.Problems on Searching and Sorting/2.Wiggle Sort.cpp
// https://leetcode.com/problems/wiggle-sort/
#include<bits/stdc++.h>
using namespace std;
// Logic-1
void wiggleSort(vector<int>& v)
{
	sort(v.begin(),v.end());
	for(int i=1;i<v.size()-1;i+=2)
		swap(v[i],v[i+1]);
}
// Logic-2
void wiggleSort(vector<int>& v)
{
	for(int i=1;i<v.size();i+=2)
	{
		if(i>0 && v[i]<v[i-1])
			swap(v[i],v[i-1]);
		if(i<v.size()-1 && v[i]<v[i+1])
			swap(v[i],v[i+1]);
	}
}
int main()
{
	int n;
	cin>>n;
	vector<int> nums;
	for(int i=0;i<n;i++)
	{
		int x;
		cin>>x;
		nums.push_back(x);
	}
	wiggleSort(nums);
	for(auto num:nums)
		cout<<num<<" ";
	return 0;
}

File: Applied Course/4.Problem Solving/2.Problems on Searching and Sorting/2.Wiggle Sort.py
# https://leetcode.com/problems/wiggle-sort/
# Logic-1
class Solution:
    def wiggleSort(self,nums):
        nums.sort()
        for i in range(1,len(nums)-1,2):
            nums[i],nums[i+1]=nums[i+1],nums[i]
        return nums
# Logic-2
class Solution:
    def wiggleSort(self,nums):
        for i in range(1,len(nums),2):
            if i>0 and nums[i]<nums[i-1]:
               nums[i],nums[i-1]=nums[i-1],nums[i]
            if i<len(nums)-1 and nums[i]<nums[i+1]:
               nums[i],nums[i+1]=nums[i+1],nums[i]
        return nums